core/impl/cluster.cxx

```diff
@@ -150,8 +149,10 @@ options_to_origin(const std::string& connection_string,
   if (opts.network.max_http_connections) {
     user_options.max_http_connections = opts.network.max_http_connections.value();
   }
-  if (!opts.network.network.empty()) {
-    user_options.network = opts.network.network;
+  user_options.network = opts.network.network;
+  if (user_options.network.empty()) {
```

Comment:
In the Public API's `couchbase::network_options`, we initialize `network_` with "auto" by default. If someone who uses the deprecated behavior.network option, doesn't also explicitly set the network.network option to the empty string, it will get ignored. The core API `network` option is also initialized with "auto" so `user_options.network` will not be empty here even without the above.

 I think we can remove the initialization of network with "auto" in the public API (as that is already the default value in the core API) and then do:
```cpp
if (!opts.network.network.empty()) {
  user_options.network = opts.network.network
} else if (!opts.behavior.network.empty()) {
  // this is deprecated option, but use it in case main options is empty
  user_options.network = opts.behavior.network
}
```

Reply:
I rather leave public API keep initializing safe value, and expect people to pass empty value if they want old setting.

Reply:
That's ok, but I think we should still change the condition to something other than `user_options.network.empty()`, since `user_options.network` is initialized to "auto" in the core as well, so it will never be empty.

I think we need something similar to the if/else-if statement from my comment above.

Reply:
The comment above is not actually true, since `user_options.network = opts.network.network` happens before that condition, so it will be empty if `opts.network.network` is empty